File: AdventOfCode2021/1/first.js
const fs = require('fs');

function getData(filename){
    const file = fs.readFileSync(filename, 'utf-8')
    const input = file.split(/\r?\n/);
    return input.map(num => parseInt(num))
}

function sumOfArray(ar){
    let sum = 0;
    ar.forEach(num => {
        sum += num
    })  
    return sum;
}
function puzzleOne(inp){
    let result = 0;
    for(let i = 1; i < inp.length; i++) {
        if( inp[i] > inp[i-1] ) {
            result += 1
        }
    }
    return result;
}
console.log("Puzzle one, first part solution: ")
console.log(puzzleOne(getData('input.txt')));

function puzzleTwo(inp){
    let result = 0;
    for(let i = 0; i < inp.length-3; i++) {
        if( sumOfArray(inp.slice(i, i+3)) < sumOfArray(inp.slice(i+1, i+4)) ) {
            result += 1
        }
    }
    return result;
}

console.log("Puzzle one, second part solution: ")
console.log(puzzleTwo(getData('input.txt')));

File: AdventOfCode2021/1/first.py
def get_data(filename):
    '''
    Read data from a file, put it in array and transform to integers.
    '''
    myfile = open(filename)
    data = myfile.read().split()
    return list(map(lambda x: int(x), data))

def puzzle_one(data):
    '''
    Loop over data and increase result for one if previous number is smaller than the next one.
    '''
    result = 0
    for i in range(1,len(data)):
        if data[i] > data[i-1]:
            result += 1
    return result

def puzzle_two(data):
    '''
    Loop over data and increase result by one if sum of three elements is bigger
    than previous sum of three digits (sum(a,b,c) < sum(b,c,d)).
    '''
    result = 0
    for i in range(0,len(data)-3):
        if sum(data[i:i+3]) < sum(data[i+1:i+4]):
            result += 1
    return result

print("Puzzle one, first part solution: ")
# If we run into error, call get_data function with full path to the file.
print(puzzle_one(get_data('input.txt')))
print("Puzzle one, second part solution: ")
# If we run into error, call get_data function with full path to the file.
print(puzzle_two(get_data('input.txt')))